### ISSUE TITLE
extra fields

### ISSUE THREAD

**Issue by hidayat-ullah** [prashants/webzash#6]
If you haven't time for Adding extra fields than Let me know specific pages for this Implement

**Comment by prashants** [comment]
You want to add cheque date and cheque number right ?

**Comment by hidayat-ullah** [comment]
absolutely

**Comment by prashants** [comment]
ok

**Comment by hidayat-ullah** [comment]
are you gonna do it ..?

**Comment by prashants** [comment]
Yes. But you will have to help - I wont be doing everything.

I will create a separate branch for it.

**Comment by hidayat-ullah** [comment]
Ya will do

**Comment by hidayat-ullah** [comment]
when  and what I have to do

**Comment by prashants** [comment]
I will let you know

**Comment by hidayat-ullah** [comment]
ok

**Comment by prashants** [comment]
I have created a separate branch for this

https://github.com/prashants/webzash/tree/extra-fields

**Comment by hidayat-ullah** [comment]
how does it work ..? mean what i hv to do

**Comment by prashants** [comment]
I will commit it today. Will message you.

**Comment by hidayat-ullah** [comment]
ok

**Comment by prashants** [comment]
Use this setup :

https://github.com/prashants/webzash/archive/extra-fields.zip

How to update

https://github.com/prashants/webzash/wiki/Webzash-update-plugin

**Comment by prashants** [comment]
Dont forget to create two fields in entries table. Check the app/Plugin/Webzash/Config/Schema.Mysql.sql for the exact schema you need to create

**Comment by hidayat-ullah** [comment]
OK thanks Lot bro..,

**Comment by hidayat-ullah** [comment]
Let me Know exact  names for DB fields

**Comment by prashants** [comment]
cheque_number cheque_date

**Comment by hidayat-ullah** [comment]
ok

**Comment by hidayat-ullah** [comment]
While im trying to install it on live server its showing the following error 
http://i.imgur.com/9QqrFhO.png

**Comment by prashants** [comment]
Which PHP version ?

**Comment by hidayat-ullah** [comment]
I Don't Know but im at Godaddy  panel

**Comment by hidayat-ullah** [comment]
its php 5.4.30

**Comment by prashants** [comment]
Do they have Sqlite support with PHP ? If not, use MySQL for user info

https://github.com/prashants/webzash/wiki/MySQL-for-user-and-account-information

**Comment by hidayat-ullah** [comment]
Ok now im at HostGator panel  will let you know if work

**Comment by hidayat-ullah** [comment]
yah it works

**Comment by prashants** [comment]
what works ?

**Comment by hidayat-ullah** [comment]
HostGator  sporting Sqlite

**Comment by hidayat-ullah** [comment]
there is an Error on print page Which Say

 Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 71 bytes) in /home3/maritime/public_html/wac_ams/lib/Cake/Event/CakeEventManager.php on line 155

**Comment by hidayat-ullah** [comment]
Any Solution ..?

**Comment by prashants** [comment]
Is download working ?

**Comment by hidayat-ullah** [comment]
Download And Print Both are Working But Along With the Error on Top..

And there is another problem.
I am unable to Recover Forgot Password Because It says (Unable to Send E-mail) 
Let me Know Where have Store the username & passwrd 
I didn't find it on Database

**Comment by prashants** [comment]
Can you send a screenshot ?

Password is stored in encrypted format in sqlite file app/Plugin/Webzash/Database/webzash.sqlite

**Comment by hidayat-ullah** [comment]
I'll send screenshot once Get Access

But How I can Replace password Because I can't Read any thing 
Its full of come Different Characters 
![untitled](https://cloud.githubusercontent.com/assets/9319496/5898248/5bab8686-a56f-11e4-9977-aaddf11391e6.png)

**Comment by prashants** [comment]
Password is encrypted.

**Comment by hidayat-ullah** [comment]
it isn't  possible to be Replaced..?

**Comment by prashants** [comment]
Dont open in notepad. Its a sqlite database ! You need to install sqlite manager plugin in firefox.

https://addons.mozilla.org/en-US/firefox/addon/sqlite-manager/

**Comment by hidayat-ullah** [comment]
Ok

**Comment by hidayat-ullah** [comment]
Thanks sir I have Done..!

current im facing another issue 

When adding a New entry and selecting a Leger form option list 
 I want the Legers Name to show on Ascending form (A-z)
Because currently its Random I think

**Comment by prashants** [comment]
Please update to new version of webzash from website.

**Comment by prashants** [comment]
I will check the random issue.

**Comment by hidayat-ullah** [comment]
is new version has Extra fields By default..?

**Comment by prashants** [comment]
I have not yet decided how to implement cheque number and date. I am still working on it.

**Comment by hidayat-ullah** [comment]
ok.!! 
By the way the version I'm currently using is fine.
Lets have a look at random issue

**Comment by prashants** [comment]
There were some bugs in it which have been fixed in the latest version.

**Comment by prashants** [comment]
Yes I will check the random issue.

**Comment by hidayat-ullah** [comment]
oK.., I'll like to use Latest version  BUT as you Know Check Number and Date is Important for us 
and its not available in New version

**Comment by prashants** [comment]
Merge the diff

**Comment by hidayat-ullah** [comment]
How.?

**Comment by prashants** [comment]
Fixed the random issue in latest commit.

**Comment by prashants** [comment]
What I suggest is you make your changes in a separate local branch. So you can easily apply those changes to latest version of master. Keep the master branch in sync with the main repo, dont make changes to master and in parallel maintain your changes in separate branch.

There are many different ways to do this. Read more about it on the net :

https://help.github.com/articles/syncing-a-fork/
http://stackoverflow.com/questions/16329776/git-howto-keep-a-branch-in-sync-with-master

**Comment by hidayat-ullah** [comment]
Dear I can't understand the methods of branches / sync
and updation is gonna be complicated for me  

Please Let me Know specific files to be replaced just for Random Issue only

**Comment by prashants** [comment]
git log --stat

**Comment by hidayat-ullah** [comment]
Or let me Know the file location Where I can Add order to Ascending

**Comment by prashants** [comment]
Its not a good thing to do the changes manually. Git has lot of tools to manage such things easily.

**Comment by hidayat-ullah** [comment]
Git mean..?

**Comment by prashants** [comment]
What os are you using ?

**Comment by hidayat-ullah** [comment]
Windows

**Comment by prashants** [comment]
https://windows.github.com/

**Comment by hidayat-ullah** [comment]
Ok So you want me to download Github on pc..?

**Comment by prashants** [comment]
Yes

**Comment by hidayat-ullah** [comment]
Why..?
is there any tool on Github to manage  webzash..?

**Comment by hidayat-ullah** [comment]
I did.

**Comment by prashants** [comment]
You need to learn a bit about git. There are lot of articles and videos on net for that.

**Comment by hidayat-ullah** [comment]
Ok I'll ..!
But I need to do this  Urgently

**Comment by prashants** [comment]
Email me what changes you made

**Comment by hidayat-ullah** [comment]
No Dear I didn't made any changes in current vrson 
 
 if i have to go with New version   I'll just need to have cheque number/ date
and Ascending Leers names only

**Comment by prashants** [comment]
Let me see what I can do

**Comment by hidayat-ullah** [comment]
ok thanks

**Comment by prashants** [comment]
https://github.com/prashants/webzash/tree/extra-fields

Merge the changes. You can fetch the updated branch here. It took me 1 sec to do it using git.

$git checkout extra-fields
$git merge master

**Comment by shadi-dabbour** [comment]
Hi I'm trying to change to 3 decimals as your instruction here:
https://github.com/prashants/webzash/wiki/Updates
but this is not working, any ideas?

**Comment by prashants** [comment]
I am planning to that in next version of Webzash. Below are the planned features. Do let me know if there is anything else that is required.

https://github.com/prashants/webzash/wiki/Webzash-Next

I will start work on it within a day or so.

**Comment by hidayat-ullah** [comment]
is there any way to use Git without command promt.?

**Comment by prashants** [comment]
Dont know for windows. You need to learn ony a few commands so it shouldnt be a issue.

**Comment by prashants** [comment]
Here is what I have planned. I will add 3 fields reference name, number and date to each ledger selection when you go to add/edit entries. It can be used for cheque, bills, invoices, etc. Is that a good idea ?

![screenshot from 2015-02-23 23 21 46](https://cloud.githubusercontent.com/assets/68140/6333411/cd7c19b6-bbb2-11e4-9c33-dd6a6c8c7cb5.png)

**Comment by hidayat-ullah** [comment]
Ya its great., But we shouldn't mention ,,reference name,, 
because it May be debitor/creditor's name if isn't then the narration field can be use for reference 
i'm saying that because no one will like a huge form 
the form should be very simple and  have less fields as possible for accounting standards.,

I think the entries form should have only two option by default ,,1 Dr and 1 Cr ,,
as we can add+ more fields on requirement
  
and the last thing is search selection 
i'm sorry to say that ,it isn't user friendly such a confusing interface for 
operators., i hope you will Agree with that points

**Comment by prashants** [comment]
I will add all the extra fields as popup so it doesnt clutter the form. I will think about the name field.

I will do something about the search.

**Comment by hidayat-ullah** [comment]
ok

**Comment by shadi-dabbour** [comment]
Hello, I can't carry forward account to next year I got this message
"Oh Snap ! Something went wrong while creating the database tables. Please check your settings and try again." but the tables are created in the database?

**Comment by prashants** [comment]
Oh can you send me the database schema that you are using.

**Comment by prashants** [comment]
Please use the setup from the website and not from github

**Comment by deamonunix** [comment]
thank you (prashants) you are the best admin, im from philippines and you have helped me a lot. from this forum...i'm willing to help other users on the future.

**Comment by deamonunix** [comment]
btw sir,  u already a code for this additional fields for cheque? thank you..

**Comment by prashants** [comment]
Not in this release. I will work on that next. Give me a week.

**Comment by deamonunix** [comment]
Thank you sir...

**Comment by prashants** [comment]
I have added a basic structure for adding cheque details and comments. Works only on add entries.

https://github.com/prashants/webzash/tree/reference

Please test this on a test database and let me know if its cool !

@hidayat-ullah, @ deamonunix

1. Download and install v2.5 of webzash
2. Create a company database.
3. Run the following SQL command on company database
https://github.com/prashants/webzash/blob/reference/PATCHES.txt
4. Copy the app/Plugin/Webzash/Database/webzash.sqlite to some place safe
5. Delete the app/Plugin/Webzash
6. Download and extract
https://github.com/prashants/webzash/archive/reference.zip
7. Rename the extracted folder to Webzash and copy it to app/Plugin folder
8. Replace the app/Plugin/Webzash/Database/webzash.sqlite file from backup

Test it and see if it solves the problem.

**Comment by deamonunix** [comment]
Good day Sir = prashants, thank.s alot, i have successfully add the reference for cheque on my database... another question is: i read the last post of older version on wenzash that you can edit the branding for reports, the question is, where i can find the Webzash.AppName for edting. thank you...

**Comment by deamonunix** [comment]
and how to auto generate the ENTRY NUMBER FIELD? thank you.

**Comment by prashants** [comment]
Branding :
https://github.com/prashants/webzash/blob/master/Config/bootstrap.php#L31

Number is by default auto-generated. You can change that in Settings > Entry Types

**Comment by prashants** [comment]
Implemented the cheque number and date and per field narration in add and edit entries.

Follow the below guide

https://github.com/prashants/webzash/wiki/Extra-fields

There is still work to be done but this is final. You can apply the patch against you main account database. Do keep a backup beforehand though.

**Comment by prashants** [comment]
I am closing this issue for now. Please follow #20 or create another issue.
